ocaml/xapi/extauth_plugin_ADpbis.ml

```diff
@@ -521,27 +523,18 @@ struct
     In addition, there are some event hooks that auth modules implement as follows:
 *)
 
-  let is_pbis_server_available max =
+  let _is_pbis_server_available max =
+    let username = "KRBTGT" in (* domain name prefix automatically added by our internal AD plugin functions *)
     let rec test i = (* let's test this many times *)
       if i > max then false (* we give up *)
       else begin (* let's test *)
         (try
-           (* (1) we _need_ to use a username contained in our domain, otherwise test (2) doesn't work *)
            (* Microsoft KB/Q243330 article provides the KRBTGT account as a well-known built-in SID in AD *)
            (* Microsoft KB/Q229909 article says that KRBTGT account cannot be renamed or enabled, making *)
            (* it the perfect target for such a test using a username (Administrator account can be renamed). *)
-           let username = "KRBTGT" in (* domain name prefix automatically added by our internal AD plugin functions *)
-           let sid = get_subject_identifier username in (* use our well-known KRBTGT builtin username in AD *)
+           ignore(get_subject_identifier username); (* use our well-known KRBTGT builtin username in AD *)
            (* OK, we found this username! *)
            debug "Request %i/%i to external authentication server successful: user %s was found" i max username;
-
-           (* (2) CA-25427: test (1) above may succeed (because of pbis caching stale AD information) *)
-           (* even though the AD domain is offline (error 32888), usually because /etc/resolv.conf is not *)
-           (* pointing to the AD server. This test should catch if the domain is offline by calling find-by-sid *)
-           (* using a domain SID. We must use a _domain_ SID. A universal SID like S-1-1-0 doesn't work for this test. *)
-           let (_: (string*string) list) = query_subject_information sid in (* use KRBTGT's domain SID *)
```

Comment:
any special reason for removing this check? otherwise, we should probably keep it to have a 2-layer defense for the domain offline problem.

Reply:
My understanding is based on the comment:
"(* (2) CA-25427: test (1) above may succeed (because of pbis caching stale AD information)"

The fix makes sure that pbis has no stale information from cache, the first check should be enough for the availability check. 
Please correct me if my understanding is wrong.
Or in reverse, is it possible that the first check passed but the second one failed?